src/cpp/session/SessionMain.cpp

```diff
@@ -3243,11 +3243,27 @@ int main (int argc, char * const argv[])
       error = workingDir.makeCurrentPath();
       if (error)
          return sessionExitFailure(error, ERROR_LOCATION);
-         
-      // start http connection listener
-      error = startHttpConnectionListener();
-      if (error)
-         return sessionExitFailure(error, ERROR_LOCATION);
+      
+      // Wait for port to be ready, if needed. This triggers when the
+      // rsession restarts but takes the port a few ms to become available
+      int retries = 10;
+      while (retries > 0)
+      {
+          // start http connection listener
+          error = startHttpConnectionListener();
+          if (error)
```

Comment:
For the sake of being conservative, how would you feel about restricting this so we only retry in the case of an `Address already in use` error, and fail immediately (as we did before) on others?